semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/nodemethods/AstNodeMethods.scala

```diff
@@ -55,4 +58,17 @@ class AstNodeMethods(val node: nodes.AstNode) extends AnyVal {
                        })
   }
 
+  def toCallOrReturn: nodes.Expression = _toCallOrReturn(node)
```

Comment:
This method would then be available on all AST nodes, right? The convention is that we drop the `to` if we traverse to another node, and use `as` as a prefix, if this is a cast. This looks like a case where we traverse to a node. It's unclear from the name though how the call or return is related to the AST node. I take it it's the call that wraps this?

Reply:
Yes we do traverse here, so `as` would be the wrong prefix.
I only refactored the code, but essentially `_toCallOrReturn` traverses to it's parent expression (e.g. call or return) by following the incoming AST nodes. It's continuing it's walk until it hits an expression that's not a generic "member access operation", e.g., "<operator>.memberAccess".

If you prefer a more generic name, maybe `toParentExpression` would work, but @ml86 may have a better idea. Just let me know what you would like.

Reply:
`parentExpression` then, without the `to`. This is consistent with e.g., `method.parameter` as opposed to `method.toParameter`.

Reply:
done